packages/authentication/src/__tests__/fixtures/helper.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Application} from '@loopback/core';
+import {RestComponent} from '@loopback/rest';
+import {AuthenticationComponent} from '../..';
+import {UserRepository} from './users/user.repository';
+
+/**
+ * Returns an application that has loaded the authentication and rest components
+ */
+export function getApp(): Application {
+  const app = new Application();
+  app.component(AuthenticationComponent);
+  app.component(RestComponent);
+  return app;
+}
+
+/**
+ * Returns a stub user repository
+ */
+export function getUserRepository(): UserRepository {
+  return new UserRepository({
+    'joe@example.com': {
+      id: '1',
+      firstname: 'joe',
+      surname: 'joeman',
+      email: 'joe@example.com',
+      password: 'joepa55w0rd',
+    },
+    'jill@example.com': {
+      id: '2',
+      firstname: 'jill',
+      surname: 'jillman',
+      email: 'jill@example.com',
+      password: 'jillpa55w0rd',
+    },
+    'jack@example.com': {
+      id: '3',
+      firstname: 'jack',
+      surname: 'jackman',
+      email: 'jack@example.com',
+      password: 'jackpa55w0rd',
+    },
+    'janice@example.com': {
+      id: '4',
+      firstname: 'janice',
+      surname: 'janiceman',
+      email: 'janice@example.com',
+      password: 'janicepa55w0rd',
+    },
+  });
+}
+
+/**
+ * Creates a Basic Authorization header value
+ *   Uses 'Basic ' as the prefix, unless another is provided
+ *   Uses ':' as a separator, unless another is provided
+ *   Can add an extra segment to create an invalid base64 string (for testing purposes)
+ */
+export interface BasicAuthorizationHeaderValueOptions {
+  alternativePrefix?: string;
+  alternativeSeparator?: string;
+  extraSegment?: string;
+}
+export function createBasicAuthorizationHeaderValue(
+  email: string,
+  password: string,
+  options?: BasicAuthorizationHeaderValueOptions,
+): string {
+  // default type is 'Basic ', unless another is specified
+  const prefix =
+    options && options.alternativePrefix ? options.alternativePrefix : 'Basic ';
+  // default separator is ':', unless another is specified
+  const separator =
+    options && options.alternativeSeparator
+      ? options.alternativeSeparator
+      : ':';
+  // extraPart is '' (blank), unless another is specified
+  const extraPart =
+    options && options.extraSegment ? separator + options.extraSegment : '';
+  return (
```

Comment:
This can be simplified as:
```ts
options = Object.assign({
  alternativePrefix: 'Basic ',
  alternativeSeparator: ':',
  extraSegment: ''
}, options);
```

Reply:
That's a great replacement for the ternary operator in this case. Thanks. :)

Reply:
Thanks for the update. Next time, please click on `Resolve conversation` if it's fixed.